### DISCUSSION TITLE
How do I get a more accurate linetype rendering for the drawing add-on

### DISCUSSION THREAD

**Discussion by radheygupta** [mozman/ezdxf#562] [Q&A] [answered]
I am trying to convert a DXF file to a black&white png file. During the conversion some lines(dotted/dashed) are drawn differently. Every line type have its special meaning so it should be drawn exactly same.
For example  in the attached screenshot DXF have dashed horizontal and vertical lines within the circle. While png version have different types of lines within the circle.

![from_dxf](https://user-images.githubusercontent.com/5222011/138216719-6f6cbc1a-a6a2-4730-b265-9e915374d5d6.png)
![from_png](https://user-images.githubusercontent.com/5222011/138216756-cc82cbfd-e5cf-40e7-9b4c-6f7968465309.png)

I have also attached the actual files for the reference.
[line_type_issue.zip](https://github.com/mozman/ezdxf/files/7386181/line_type_issue.zip)

**Comment by mozman** [comment, marked as answer]
The default rendering mode for line types in the `drawing` add-on is `APPROIXIMATE`, which uses the line type rendering provided by the backend. `Matplotlib` and `PyQt` render line types in such a way that the same line pattern is always displayed regardless of the view scaling or zoom level. The `ACCURATE` line policy is implemented for both backends, which shows a more precise rendering. The modified code uses the new configuration system for the `drawing` add-on and requires `ezdxf` v0.17 run:

```Python
from pathlib import Path

import matplotlib.figure as figure
import ezdxf
from ezdxf.addons.drawing import RenderContext, Frontend
from ezdxf.addons.drawing.matplotlib import MatplotlibBackend
from ezdxf.addons.drawing.config import Configuration, LinePolicy

# set your working directory:
DIR = Path("~/Desktop/Now/ezdxf/561").expanduser()

def convert_dxf2img(name: Path):

    doc = ezdxf.readfile(name)
    msp = doc.modelspace()

    # new configuration system for the drawing add-on in ezdxf v0.17:
    config = Configuration.defaults()
    config = config.with_changes(line_policy=LinePolicy.ACCURATE)

    fig = figure.Figure()
    ax = fig.add_axes([0, 0, 1, 1])
    ctx = RenderContext(doc)
    ctx.set_current_layout(msp)
    ctx.current_layout_properties.set_colors(bg="#ffffff", fg="#000000")
    out = MatplotlibBackend(ax)
    frontend = Frontend(ctx, out, config=config)
    frontend.draw_layout(msp, finalize=True)
    fig.savefig(name.with_suffix(".png"), dpi=300)


if __name__ == "__main__":
    convert_dxf2img(DIR / "MBV1G-PB0000047471-2012-06-01-DE-3.dxf")
```
![MBV1G-PB0000047471-2012-06-01-DE-3](https://user-images.githubusercontent.com/4230966/138296630-7ae3ddd5-c538-456d-9f38-d6014f7eea6b.png)
